<div align="center"><i>Tsute Chen (George)</i><br>
  The Forsyth Institute, Cambridge, MA, USA</div>

## Part I. Instroduction - how to use this workshop tutorial

### 1. R Code Box
The text in the light-blue box, contains the R codes that you can highlight (by dragging your mouse), copy (use the control-C key combination) and paste (control-V) into your R command-line interface on your computer. For example:
``` R
#Below is a working R code that you can copy and paste into your R to execuate some R commands
ls()
message("Hello! Welcome to the RESISTPART Bioinformatics Workshop")
```

### 2. Alternative method box

Text inside the alternative method box, shows another or more different ways of doing the same thing:

<table bgcolor="#C8FBD3"><tr><td style="font-size:0.8em" width="100%">
<b>Alternative:</b><br>
This is where other ways of doing the same thing will be described
</td></tr></table>

### 3. Notes and comments

<table  bgcolor="#DDDDDD"><tr><td style="font-size:0.8em;font-style:italic;">
  <b>Note:</b> A note is something that needs to be mentioned but is apart from the context.
</td></tr></table>



<br><br><br>

## Part II. Workshop Tutorial

#### Analysis of 16S rRNA gene sequences for microbiome diversity study

### 1. Introduction

The goal of this workshop is to gain hands-on experience for analyzing the 16S rRNA gene sequences generated by the next generation sequencing (NGS) platform, such as the Illumina sequencers. The 16S rRNA gene is commonly used by the research community as a marker to decipher the diversity of the microbial community in a sample. The reasons for this are first, 16S rRNA gene is universally present in all prokaryotes (inlcuding Bacteria and Archaea); secondly, although the 16S rRNA gene sequences are very conserved in both Bacteria and Archaea, there are enough variability within the genes (i.e,. 9 hypervariable regions) for different species. Hence by sequencing the 16S rRNA genes in a microbial community, and matchinng the sequences to a set of 16S rRNA gene sequences with know taxonomy information, one can determine how many and what are the species in the samples. Since the NGS has a limitation of how long a DNA sequence can be decoded, only a portion of the 16S rRNA gene can be targeted. Usually the first step is to PCR-amplify a region of the 16S rRNA gene from the DNA extracted from the samples. The amplicons are then sequenced by the NGS technology to produce millions of DNA reads. These sequence reads are then filtered and denoised to reduce the errors and artifacts. The filtered reads are then taxonomically aasigned to determine their genus or species. The outcome of these processes is a read count table with columns representing samples and rows representing organisms, either genus, species or OTUs - operational taxonomic units, if the taxnomy has yet to be determined). This table is usually called an OTU table.


### 2. Installation of the R software and packages for analyzing microbiome sequence data

### 3. Basic R operations

```R
#This is R code box
ls()
save.image()
```

### 4. Use of the R DADA2 package to process the 16S rRNA gene sequence reads

### 5. Use of the R Phyloseq package to study microbial diversity

### 6. Use of the MicrobiomeAnalyst online tools for statistical, visual and meta-analysis of microbiome data

Link to Archal's MicrobiomeAnalyst tutorial

### 7. Beyond microbiome sequence data â€“ Meta-genomic and Meta-transcriptomic data


```R
#This is R code box
ls()
save.image()
```
