<div align="center">By <i>Tsute George Chen</i>, The Forsyth Institute, Cambridge, MA, USA</div>

## Part I. Instruction on how to use this workshop Toturial

### I.1. R Code Box
The text in the light-gray box contains the R codes that you can highlight, copy and paste into your R command-line interface.
For example:
``` R
#Below is a working R code that you can copy and paste into your R to execuate some R commands
ls()
message("Hello! Welcome to the RESISTPART Bioinformatics Workshop")
```

### I.2. Alternative method box

Text inside the alternative method box show another or more different ways of doing the same thing:

<table><tr><td>
<table style="text-align:left;background-color:#EEEEEE;font-size:0.8em">
<tr><td width="100%" bgcolor="#99CCFF">Alternative</td></tr>
<tr><td>This is where other ways of doing the same thing will be described.</td></tr></table>
</td><tr></table>

### I.3. Notes and comments

**Note:** a note is something that needs to be mentioned but is apart from the context.

## Part II. Workshop Tutorial

Something here
### 1. Introduction

### 2. Installation of the R software and packages for analyzing microbiome sequence data

### 3. Basic R operations

```R
#This is R code box
ls()
save.image()
```

### 4. Use of the R DADA2 package to process the 16S rRNA gene sequence reads

### 5. Use of the R Phyloseq package to study microbial diversity

### 6. Use of the MicrobiomeAnalyst online tools for statistical, visual and meta-analysis of microbiome data

Link to Archal's MicrobiomeAnalyst tutorial

### 7. Beyond microbiome sequence data â€“ Meta-genomic and Meta-transcriptomic data


```R
#This is R code box
ls()
save.image()
```
